tempodb/encoding/vparquet2/block_traceql.go

```diff
@@ -55,21 +55,41 @@ func (s *span) DurationNanos() uint64 {
 
 func (s *span) DescendantOf(x traceql.Span) bool {
 	if ss, ok := x.(*span); ok {
-		return s.nestedSetLeft > ss.nestedSetLeft && s.nestedSetLeft < ss.nestedSetRight
+		if s.nestedSetLeft == 0 ||
+			s.nestedSetRight == 0 ||
+			ss.nestedSetLeft == 0 ||
+			ss.nestedSetRight == 0 {
+			// Spans with missing data, never a match.
+			return false
+		}
+		return s.nestedSetLeft > ss.nestedSetLeft && s.nestedSetRight < ss.nestedSetRight
```

Comment:
The typo was the third var in this statement `s.nestedSetLeft` -> `s.nestedSetRight`